app/javascript/src/components/Dashboard/Notes/constants.js

```diff
@@ -48,6 +45,32 @@ export const ROLES = [
   },
 ];
 
+export const NOTES_FORM_VALIDATION_SCHEMA = yup.object().shape({
+  title: yup.string().required("Title is required"),
+  description: yup.string().required("Description is required"),
+  role: yup
+    .object()
+    .nullable()
+    .shape({
+      label: yup.string().oneOf(ROLES.map(tag => tag.label)),
+      value: yup.string().oneOf(ROLES.map(tag => tag.value)),
+    })
+    .required("Role is required"),
+  tags: yup
+    .array(
+      yup
+        .object()
+        .nullable()
+        .shape({
+          label: yup.string().oneOf(TAGS.map(tag => tag.label)),
+          value: yup.number().oneOf(TAGS.map(tag => tag.value)),
+        })
+    )
+    .nullable()
```

Comment:
I believe this `nullable` is added over here since you set up the initial value as `null`. If we change it to `[]` then I don't think nullable is needed here.